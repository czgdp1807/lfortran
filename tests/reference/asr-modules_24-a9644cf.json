{
    "basename": "asr-modules_24-a9644cf",
    "cmd": "lfortran --indent --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/modules_24.f90",
    "infile_hash": "aeaa1e42330e40c5f303e6d25e54e5bb7fa38bfff6d49cf3491f10fb",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-modules_24-a9644cf.stdout",
    "stdout_hash": "f0907ec47e6f82f30e416b25fff65dccf93df8a47f2b515844215bc0",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}