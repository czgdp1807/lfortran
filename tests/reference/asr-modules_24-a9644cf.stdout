(TranslationUnit
    (SymbolTable
        1
        {
            tomlf_de_tokenizer:
                (Module
                    (SymbolTable
                        2
                        {
                            parse_select:
                                (Function
                                    (SymbolTable
                                        5
                                        {
                                            de:
                                                (Variable
                                                    5
                                                    de
                                                    []
                                                    InOut
                                                    ()
                                                    ()
                                                    Default
                                                    (Class
                                                        2 toml_tokenizer
                                                        []
                                                    )
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            table:
                                                (Variable
                                                    5
                                                    table
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Pointer
                                                        (Struct
                                                            2 toml_table
                                                            []
                                                        )
                                                    )
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    parse_select
                                    []
                                    [(Var 5 de)]
                                    [(Nullify
                                        [5 table]
                                    )
                                    (=>
                                        (StructInstanceMember
                                            (Var 5 de)
                                            4 current
                                            (Pointer
                                                (Struct
                                                    2 toml_table
                                                    []
                                                )
                                            )
                                            ()
                                        )
                                        (Var 5 table)
                                    )]
                                    ()
                                    Source
                                    Public
                                    Implementation
                                    ()
                                    .false.
                                    .false.
                                    .false.
                                    .false.
                                    .false.
                                    []
                                    []
                                    .false.
                                ),
                            toml_table:
                                (StructType
                                    (SymbolTable
                                        3
                                        {
                                            inline:
                                                (Variable
                                                    3
                                                    inline
                                                    []
                                                    Local
                                                    (LogicalConstant
                                                        .false.
                                                        (Logical 4 [])
                                                    )
                                                    ()
                                                    Save
                                                    (Logical 4 [])
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    toml_table
                                    []
                                    [inline]
                                    Source
                                    Public
                                    .false.
                                    ()
                                    ()
                                ),
                            toml_tokenizer:
                                (StructType
                                    (SymbolTable
                                        4
                                        {
                                            current:
                                                (Variable
                                                    4
                                                    current
                                                    []
                                                    Local
                                                    (PointerNullConstant
                                                        (Pointer
                                                            (Struct
                                                                2 toml_table
                                                                []
                                                            )
                                                        )
                                                    )
                                                    ()
                                                    Default
                                                    (Pointer
                                                        (Struct
                                                            2 toml_table
                                                            []
                                                        )
                                                    )
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    toml_tokenizer
                                    [toml_table]
                                    [current]
                                    Source
                                    Public
                                    .false.
                                    ()
                                    ()
                                )
                        })
                    tomlf_de_tokenizer
                    []
                    .false.
                    .false.
                )
        })
    []
)
